pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -531,7 +561,8 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 	return reconcile.Result{}, nil
 }
 
-// setDefaultsOnIntrusionDetection updates the IntrusionDetection resource with defaults if ComponentResources is not populated.
+// setDefaultsOnIntrusionDetection updates the IntrusionDetection resource with defaults if
+// ComponentResources and AnomalyDetectionSpec are not populated.
 func (r *ReconcileIntrusionDetection) setDefaultsOnIntrusionDetection(ctx context.Context, ids *operatorv1.IntrusionDetection) error {
```

Comment:
nit: I believe similar functions in other controllers are named "fillDefaults(...)"  and the updating is then done from the reconcile() func.
For example: `logstorage_controller.go`